cmd/xl-v1-multipart.go

```diff
@@ -416,6 +416,19 @@ func (xl xlObjects) PutObjectPart(bucket, object, uploadID string, partID int, s
 	// Delete the temporary object part. If PutObjectPart succeeds there would be nothing to delete.
 	defer xl.deleteObject(minioMetaBucket, tmpPartPath)
 
+	if size > 0 {
+		for _, disk := range onlineDisks {
+			if disk != nil {
+				sizeInDisk := size / int64(xlMeta.Erasure.DataBlocks)
```

Comment:
exact sizeInDisk will be:
```
numBlocks := size/xlMeta.Erasure.BlockSize

chunkSize := getChunkSize(xlMeta.Erasure.BlockSize, dataBlocks)

sizeInDisk := numBlocks * chunkSize

remaining := size%xlMeta.Erasure.BlockSize
if remaining > 0 {
    sizeInDisk += getChunkSize(remaining, dataBlocks)
}
```

But instead of having the above complicated calculation we can take an approximate value of :
`sizeInDisk := size / int64(xlMeta.Erasure.DataBlocks)`

this is good enough for fallocate. You can add a comment saying this is an approx. value not the exact value.

and delete the following as it is not correct:
```
if size%int64(xlMeta.Erasure.DataBlocks) > 0 {
    sizeInDisk++
}
```